### ISSUE TITLE
Per-method parameterization

### ISSUE THREAD

**Issue by jskeet** [dotnet/BenchmarkDotNet#256]
This is related to #60.

I've read the [docs on parameterization](https://perfdotnet.github.io/BenchmarkDotNet/Advanced/Params.htm) but this seems a little clunky compared with what I'm used to for unit tests. If I want to benchmark several methods, with different sets of parameters, it's annoying to need a separate class for each method.

For unit tests, I can specify constants via attributes, and sources of parameters by name. It would be nice to have this in unit tests too, e.g.

``` csharp
[Benchmark]
[BenchmarkCase(10)]
[BenchmarkCase(50)]
public void Foo(int x)
{
...
}
```

Or:

``` csharp
private static readonly int[] FooValues = DoSomeComputationBecauseItsNotConstant();

[Benchmark]
[BenchmarkCaseSource(nameof(FooValues))]
public void Foo(int x)
{
...
}
```

Apologies if this is a duplicate...

**Comment by adamsitnik** [comment]
I am going to work on some PoC

**Comment by Joe4evr** [comment]
> ```cs
> private static readonly int[] FooValues = DoSomeComputationBecauseItsNotConstant();
> ```

Or something that *could* be constant, but it's *a lot*, say, `Enumerable.Range(1, 1000).ToArray()`. 🙃

**Comment by adamsitnik** [comment]
@Joe4evr it's already implemented, currently being reviewed in #577 , should be released as part of 0.10.11

**Comment by AndreyAkinshin** [comment]
@adamsitnik, hooray!
